app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitItemAdapter.java

```diff
@@ -15,36 +14,43 @@
 import com.example.habitsmasher.Habit;
 import com.example.habitsmasher.HabitList;
 import com.example.habitsmasher.R;
-
-import java.text.SimpleDateFormat;
-import java.util.ArrayList;
-import java.util.Locale;
+import com.firebase.ui.firestore.FirestoreRecyclerAdapter;
+import com.firebase.ui.firestore.FirestoreRecyclerOptions;
+import com.firebase.ui.firestore.ObservableSnapshotArray;
 
 /**
  * The habit item adapter is the custom adapter for the RecyclerView that holds the habit list
  */
-public class HabitItemAdapter extends RecyclerView.Adapter<HabitItemAdapter.HabitViewHolder> {
-    private final Context _context;
+public class HabitItemAdapter extends FirestoreRecyclerAdapter<Habit, HabitItemAdapter.HabitViewHolder> {
+    private Context _context;
     private static HabitList _habits;
+    private final ObservableSnapshotArray<Habit> _snapshots;
```

Comment:
This is the array of snapshots from the database. It holds all the current documents in the Firebase db.